pkg/connection/connection.go

```diff
@@ -78,9 +78,9 @@ func (conn ApiConnection) Do(request *http.Request) ([]byte, error) {
 		return nil, err
 	}
 
-	if !successCode(response.StatusCode) {
-		msg := parseError(response.Body)
-		return nil, fmt.Errorf("API error: %v (code: %v)", msg, response.StatusCode)
+	// Check if there was an error from the given API response.
+	if api_error := ErrorFromResponse(response); api_error != nil {
+		return nil, api_error
```

Comment:
```suggestion
	// Check if there was an error from the given API response.
	if apiError := ErrorFromResponse(response); apiError != nil {
		return nil, apiError
```

somebody is in the C syntax world I see :P